src/ng-multiselect-dropdown/test/multi-select.component.spec.ts

```diff
@@ -157,6 +157,20 @@ describe('ng-multiselect-component', function() {
       tickAndDetectChanges(fixture);
       expect(fixture.componentInstance.select.isDropdownOpen).toBe(false);
     }));
+
+    it('should not select disabled item', fakeAsync(() => {
```

Comment:
how this test case is getting passed. You do not have a single item with isDisabled true.

Reply:
okay, you have explicitly disabled it(line no 163). my bad :)

Reply:
There's an issue with type in line 163 if no "isDisabled" property is defined for the elements in the cities array, since it is of implicit type. To delete all those isDisabled properties, I would have to define a type for those values, declare the array with that explicit type and then omit those properties.

something around the works of:
type Item = {
  item_id: number,
  item_text: string,
  isDisabled?: boolean
};

and then:

cities: Item[] = [
    { item_id: 1, item_text: 'Mumbai' },
    { item_id: 2, item_text: 'Bangalore' },
    { item_id: 3, item_text: 'Pune' },
    { item_id: 4, item_text: 'Navsari' },
    { item_id: 5, item_text: 'New Delhi' }
];

Reply:
I just didn't want to declare a new type for the test and assumed adding the property to each element was easier.

Reply:
> 
> 
> okay, you have explicitly disabled it(line no 163). my bad :)
> I would suggest you to have a item with true flag, do not disable it explicitly. this way, sample code will showcase how disabled flag behaves.

So, do you want me to declare a type for those items and set one to true? Or should I leave those isDisabled properties there and just set one to true?

Reply:
Also, want me to add changes to the demo so it shows this new feature?

Reply:
please add to the demo. is it good practice to merge PR with 1 commit. can you please rebase your all commit?

Reply:
I added the feature to the multiple-demo and also squashed all commits into one.